src/blueprint/from_config/operators/call.rs

```diff
@@ -0,0 +1,79 @@
+use crate::blueprint::*;
+use crate::config;
+use crate::config::{Config, Field, GraphQLOperationType};
+use crate::directive::DirectiveCodec;
+use crate::try_fold::TryFold;
+use crate::valid::Valid;
+
+pub fn update_call(
+  operation_type: &GraphQLOperationType,
+) -> TryFold<'_, (&Config, &Field, &config::Type, &str), FieldDefinition, String> {
+  TryFold::<(&Config, &Field, &config::Type, &str), FieldDefinition, String>::new(
+    move |(config, field, type_of, name), b_field| {
+      let Some(call) = &field.call else {
+        return Valid::succeed(b_field);
+      };
+
+      if validate_field_has_resolver(name, field, &config.types).is_succeed() {
+        return Valid::fail(format!(
+          "@call directive is not allowed on field {} because it already has a resolver",
+          name
+        ));
+      }
+
+      let type_and_field = if let Some(query) = &call.query {
+        Valid::succeed(("Query", query.as_str()))
+      } else {
+        Valid::fail("call must have query".to_string())
+      };
+
+      type_and_field
+        .and_then(|(type_name, field_name)| {
+          Valid::from_option(
+            config.find_type(type_name),
+            format!("{} type not found on config", type_name),
+          )
+          .zip(Valid::succeed(field_name))
+        })
+        .and_then(|(query_type, field_name)| {
+          Valid::from_option(
+            query_type.fields.get(field_name),
+            format!("{} field not found", field_name),
+          )
+          .and_then(|field| {
+            if field.has_resolver() {
+              Valid::succeed((field, field_name, call.args.iter()))
+            } else {
+              Valid::fail(format!("{} field has no resolver", field_name))
+            }
+          })
+        })
+        .and_then(|(field, field_name, args)| {
+          args.fold(Valid::succeed(field.clone()), |field, (key, value)| {
+            field.and_then(|field| {
+              let value = value.replace("{{", "").replace("}}", "");
+
+              if let Some(http) = field.clone().http.as_mut() {
+                http.path = http.path.replace(format!("args.{}", key).as_str(), value.as_str());
```

Comment:
replacing string parts is too intrusive and brittle. Let's think about some way to be able to override args for existing resolver i.e. some flags or wrappers that will override args resolution

Reply:
Do we even need to replace this? I think just cloning the resolves should do the job.

Reply:
agree 100% with you that this solution is at minimun sub-optimal, but I'll resolve this and we can continue the discussion on alternatives on [this comment](https://github.com/tailcallhq/tailcall/pull/881#discussion_r1440691499) responses.